### PR TITLE
Use System.nanoTime() to initialize Engine.lastWriteNanos

### PR DESCRIPTION
**Pull request by s1monw**
On _lastWriteNanos_ we use System.nanoTime() to initialize this since:
- we use the value for figuring out if the shard / engine is active so if we startup and no write has happened yet we still consider it active
  for the duration of the configured active to inactive period. If we initialize to 0 or Long.MAX_VALUE we either immediately or never mark it
  inactive if no writes at all happen to the shard.
- we also use this to flush big-ass merges on an inactive engine / shard but if we we initialize 0 or Long.MAX_VALUE we either immediately or never
  commit merges even though we shouldn't from a user perspective (this can also have funky sideeffects in tests when we open indices with lots of segments
  and suddenly merges kick in.

### CODE REVIEW

**Comment by s1monw** [comment]
@mikemccand can you look

**Comment by mikemccand** [comment]
LGTM, thanks @s1monw!

File: core/src/main/java/org/elasticsearch/index/engine/Engine.java [hunk 1]
```diff
@@ -80,7 +80,18 @@
     protected final ReleasableLock readLock = new ReleasableLock(rwl.readLock());
     protected final ReleasableLock writeLock = new ReleasableLock(rwl.writeLock());
     protected volatile Throwable failedEngine = null;
-    protected volatile long lastWriteNanos = Long.MAX_VALUE; // no write yet!
+    /*
+     * on <tt>lastWriteNanos</tt> we use System.nanoTime() to initialize this since:
+     *  - we use the value for figuring out if the shard / engine is active so if we startup and no write has happened yet we still consider it active
```
Comment (bleskes) [inline core/src/main/java/org/elasticsearch/index/engine/Engine.java]: minor note: we consider shards inactive until the first indexing operation has happen, so I think this part is OK regardless of the change.

**Comment by bleskes** [comment]
LGTM2

**Pull request merged**